### PR TITLE
feat: simplify impl of set_host and set_hostname

### PR DESCRIPTION
**Pull request by debadree25**
Have moved the implementation to a private helper within the URL struct, I tried moving it to within helpers but was causing issues since the involves updating fields within the object.

Fixes: https://github.com/ada-url/ada/issues/131

### CODE REVIEW

File: src/url-setters.cpp [hunk 1]
```diff
@@ -151,70 +151,12 @@ namespace ada {
...
-      // Set url’s host to the empty string.
-      host = "";
-    }
-    else {
-      // Let host be the result of host parsing buffer with url is not special.
-      if (!parse_host(new_host)) {
-        host = previous_host;
-        return false;
-      }
+  bool url::set_host(const std::string_view input) {
```
Comment (anonrig) [inline src/url-setters.cpp:154]: This is a breaking change. We should not remove `set_host` or `set_hostname` as a function, but keep the API as it is and just simplify by calling a function similar to what you proposed.

File: include/ada/url.h [hunk 2]
```diff
@@ -366,6 +366,16 @@ namespace ada {
...
     [[nodiscard]] bool parse_opaque_host(std::string_view input);
 
+    /**
+     * @private
+     * 
+     * Sets the host or hostname according to override condition.
+     * Return true on success.
+     * @see https://url.spec.whatwg.org/#hostname-state
+     */
+
```
Comment (anonrig) [inline include/ada/url.h:376]: Unnecessary newline

File: include/ada/url.h [hunk 3]
```diff
@@ -366,6 +366,16 @@ namespace ada {
...
+    [[nodiscard]] bool set_host_or_hostname(std::string_view input, bool override_hostname);
```
Comment (anonrig) [inline include/ada/url.h:377]: Please move this function below the `setters` in this file.

**Review by anonrig — CHANGES_REQUESTED** [review]

File: src/url-setters.cpp (hunk 1 above)
Comment (debadree25) [reply src/url-setters.cpp:154]: I think the git diff is making it a little confusing 😅 set_host and set_hostname are just calling set_host_or_hostname now as a helper, I havent removed the setters

**Comment by lemire** [comment]
I think that this is quite nice. Thanks @debadree25 !!!

File: src/url-setters.cpp (hunk 1 above)
Comment (anonrig) [reply src/url-setters.cpp:154]: Yeah my bad! Git diff is quite confusing. Resolving this.

**Review by anonrig — APPROVED** [review]

File: include/ada/url.h (hunk 3 above)
Comment (debadree25) [reply include/ada/url.h]: have moved it
also, one question would it also not make this function not private? is that ok? (my cpp knowledge isnt the sharpest 😅😅)

File: include/ada/url.h (hunk 2 above)
Comment (debadree25) [reply include/ada/url.h]: removed

**Pull request merged**